cmd2.py

```diff
@@ -549,17 +538,18 @@ def cmd_wrapper(instance, cmdline):
         # we want it to be the name of our command
         argparser.prog = func.__name__[3:]
 
-        # put the help message in the method docstring
-        if func.__doc__:
+        # If the description has not been set, then use the method docstring if one exists
+        if not argparser.description and func.__doc__:
             argparser.description = func.__doc__
 
         cmd_wrapper.__doc__ = argparser.format_help()
 
         # Mark this function as having an argparse ArgumentParser (used by do_help)
         cmd_wrapper.__dict__['has_parser'] = True
 
-        # If there are subcommands, store their names to support tab-completion of subcommand names
-        if subcommand_names is not None:
+        # If there are subcommands, store their names in a list to support tab-completion of subcommand names
+        if argparser._subparsers is not None:
```

Comment:
It would be nice if ``argparse`` made this easier, but neither @kotfu or I could find a good way.  So thanks for this.